main.c

```diff
@@ -29,57 +33,95 @@ void printString (char* str) {
 }
 
 void saveCursorPos () {
-    printf("%c7", 27);
-    fflush(stdout);
+    printSpecialChar("7");
 }
 
 void restoreCursorPos () {
-    printf("%c8", 27);
-    fflush(stdout);
+    printSpecialChar("8");
+}
+
+void redrawPrevLines (char matrix[1000][1000], int currentLine)
+{
+  int i, j;
+  int width = w.ws_col;
+  saveCursorPos();
+  gotoxy(0, 0);
+  clearLine();
+  for (i = 1; i <= currentLine; i++){
+    gotoxy(0, i);
+    clearLine();
+    for(j = 1; j <= width; j++) {
+      gotoxy(j, i);
+      printf("%c", matrix[i][j]);
+    }
+  }
+  restoreCursorPos();
 }
 
 void *drawProgress(void *null)
 {
     int width = w.ws_col;
```

Comment:
width is never changed, please make it const

Reply:
Done